crates/driver/src/domain/competition/mod.rs

```diff
@@ -103,6 +103,37 @@ impl Competition {
             }
         });
 
+        let solutions = solutions.filter(|solution| {
+            let mut amm_orders = HashMap::<eth::Address, usize>::new();
+            for trade in solution.trades() {
+                let owner = match trade {
+                    solution::Trade::Fulfillment(f) => {
+                        let order = auction
+                            .orders()
+                            .iter()
+                            .find(|o| o.uid == f.order().uid)
+                            .unwrap();
+                        order.trader().0
+                    }
+                    solution::Trade::Jit(j) => j.order().signature.signer,
+                };
+                let is_cow_amm_order = self.eth.contracts().cow_amms().contains(&owner);
+                if is_cow_amm_order {
+                    let entry = amm_orders.entry(owner).or_default();
+                    *entry += 1;
+                    if *entry > 1 {
```

Comment:
We can probably replace `HashMap::<eth::Address, usize>` with `HashSet::<eth::Address>`

Reply:
True. Initially collected the total counts in the map and returned false when some count was greater than 1.